src/main/listeners.ts

```diff
@@ -43,4 +50,26 @@ export function setupListeners() {
       return { path, valid: false };
     }
   });
+
+  ipc_getDesktopAppPath.main!.handle(async () => {
+    // get the path and check existence
+    const desktopAppPath = path.join(app.getPath("appData"), "Slippi Desktop App");
+    const exists = await fs.pathExists(desktopAppPath);
+
+    return { path: desktopAppPath, exists: exists };
+  });
+
+  ipc_deleteFolder.main!.handle(async ({ path }) => {
+    await fs.remove(path).catch((err) => {
+      throw new Error(err);
+    });
+    return { success: true };
+  });
+
+  ipc_getFolderContents.main!.handle(async ({ path }) => {
+    const contents = await fs.readdir(path, { withFileTypes: true }).catch((err) => {
+      throw new Error(err);
+    });
+    return { success: true, contents };
+  });
```

Comment:
This kind of very general purpose things should be avoided, specially the `deleteFolder` call. IPC calls should be more specific to a single task. We can think of the front end as the client, which we shouldn't trust. If the frontend happens to load a rogue script, it could call `ipc_deleteFolder.renderer!.trigger({path: "C:\\Windows"})` for example. The IPC calls should be much more specific for example:
```ts
await ipc_handleMigration.renderer!.trigger({
  migrateNetplay: "C:\\Some\\Path",
  migratePlayback: false,
})
```

This way even if the frontend goes rogue, their scope is only limited to calling that one IPC function.

Obviously having `nodeIntegration: true` bypasses all of this security anyways, but it's still best practice to have specific IPC functions.